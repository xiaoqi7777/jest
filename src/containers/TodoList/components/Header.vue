<template>
  <div :class="$style.header">
    <div :class="$style.headerContent">
      TodoList1111
      <input
      :class="$style.headerInput"
        data-test='input'
        v-model='inputValue'
        @keyup.enter='addTodoItem'
        placeholder="TodoList!"
        />
      </div>
  </div>
</template>

<script>
export default {
  data () {
    return {
      inputValue: ''
    }
  },
  methods: {
    addTodoItem () {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.inputValue = ''
      }
    }
  }
}
</script>

<style  module>
.header{
  line-height: 60px;
  background :#666;
}
.headerContent{
  width:600px;
  margin: 0 auto;
  color: #fff;
  font-size: 24px;
}
.headerInput{
  float: right;
  width: 260px;
  margin-top: 12px;
  line-height: 24px;
  outline: none;
  color:#333;
  text-indent: 10px;
}
</style>
