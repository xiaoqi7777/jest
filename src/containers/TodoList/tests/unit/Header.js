/* eslint-disable no-undef */
// 内部自带的
// import { shallowMount, mount } from '@vue/test-utils'
import Header from '../../components/Header.vue'
// import { findTestWrapper } from '../../../../utils/testUtils'
// it('Header 样式发生改变 做提示', () => {
//   const wrapper = shallowMount(Header)
//   // 处理样 监听样式的变化 给出提示
//   expect(wrapper).toMatchSnapshot()
// })
describe('11', () => {
  const wrapper = mount(Header)
  it('1', () => {
    expect(wrapper.html()).toContain('<div>test</div>')
    expect(wrapper.find('input')).toEqual({ selector: 'input' })
    expect(wrapper.vm.inputValue).toBe('')
    const button = wrapper.find('input')
    button.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeFalsy()
  })
})

// it('Header 包含 input 框', () => {
//   const wrapper = shallowMount(Header)
//   console.log('****', wrapper)
//   const input = wrapper.find('[data-test="input"]')
//   // exists是否存在
//   expect(input.exists()).toBe(true)
// })

// it('Header 中 input 框初始内容为空', () => {
//   const wrapper = shallowMount(Header)
//   const inputValue = wrapper.vm.$data.inputValue
//   // exists是否存在
//   expect(inputValue).toBe('')
// })

// it('Header 中 input 框值发生变化,数据应该跟着变', () => {
//   const wrapper = shallowMount(Header)
//   const input = findTestWrapper(wrapper, 'input')
//   input.setValue('dell lee')
//   const inputValue = wrapper.vm.$data.inputValue
//   expect(inputValue).toBe('dell lee')
// })

// it('Header 中 input 框输入回车,无内容时,无反应', () => {
//   const wrapper = shallowMount(Header)
//   const input = findTestWrapper(wrapper, 'input')
//   input.setValue('')
//   input.trigger('keyup.enter')
//   // 当数据为空 回撤的时候 事件不被触发
//   expect(wrapper.emitted().add).toBeFalsy()
// })

// it('Header 中 input 框输入回车,有内容时,向外触发事件,同时情况 inputValue', () => {
//   const wrapper = shallowMount(Header)
//   const input = findTestWrapper(wrapper, 'input')
//   input.setValue('dell lee')
//   input.trigger('keyup.enter')
//   // 当有数据 回撤的时候 事件被触发 同时情况数据
//   expect(wrapper.emitted().add).toBeTruthy()
//   expect(wrapper.vm.$data.inputValue).toBe('')
// })
