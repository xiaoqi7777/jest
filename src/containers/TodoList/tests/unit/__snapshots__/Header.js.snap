// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 样式发生改变 做提示 1`] = `
<div class="header">
  <div class="headerContent">
    TodoList1111
    <input data-test="input" placeholder="TodoList!" class="headerInput"></div>
</div>
`;
